actions.js

```diff
@@ -1,157 +1,391 @@
-exports.getActions  = function() {
+const { InstanceBase, TCPHelper } = require('@companion-module/base')
+const nova_config = require('./config');
 
-	let actions = {};
+// calculates the checksum that is required when sending a bytestream for layer configuration to the VX1000
+function getLayerUpdateCommandVX1000Checksum (pipCommandBuffer)
+{
+	// we can't include the first two bytes in our checksum calculation, so we are taking a subarray excluding the first two bytes
+	const summableBuffer = pipCommandBuffer.subarray(2);
+
+	var sum = 0;
+
+	// sum up all of the bytes (except for the first two bytes)
+	for (let i = 0; i < summableBuffer.length; i++)
+	{
+		sum += summableBuffer[i];
+	}
+
+	// add the magic number (from novastar) to the end of the sum to generate the checksum
+	sum += 0x5555;
+
+	// split the sum into two bytes in little endian, this will be placed on the end of our command to serve as the checksum for the VX1000 to verify
+	const resultChecksumBuffer = Buffer.allocUnsafe(2);
+	resultChecksumBuffer.writeInt16LE(sum);
+	
+	return resultChecksumBuffer;
+}
+
+// function to build the byte sequence given the configurable layer options for configuring layers on VX1000
+function getLayerUpdateCommandVX1000 (enabled, initialX, initialY, hWidth, vHeight, layerBuffer, cardSlotBuffer, layerPriorityBuffer, 
+	connectorCodeBuffer, opacity)
+{
+	var bufferArray = [];
+
+	// the initial header of the command
+	bufferArray.push(Buffer.from([0x55,0xaa,0x00,0x00,0xfe,0x00,0x00,0x00,0x00,0x00,0x01,0x00,0x40,0x00,0x02,0x13,0x30,0x00]));
+
+	// whether the layer is enabled or not
+	bufferArray.push(enabled == '1' ? Buffer.from([0x01]) : Buffer.from([0x00]));
+
+	// layer number
+	bufferArray.push(layerBuffer);
+	
+	// CardNo
+	bufferArray.push(cardSlotBuffer);
+
+	// Priority
+	bufferArray.push(layerPriorityBuffer);
+
+	// Source (connector code)
+	bufferArray.push(connectorCodeBuffer);
+
+	// initial X position
+	const initialXBuffer = Buffer.allocUnsafe(4);
+	initialXBuffer.writeInt32LE(initialX)
```

Comment:
write will fail if initialX is no integer in valid range. You should round it to integer here and clamp the range in the action option.
Same for the other buffer assignments.